native_client/javascript/Makefile

```diff
@@ -42,7 +42,9 @@ node-wrapper: copy-deps build
 	$(NODE_BUILD_TOOL) $(NODE_PLATFORM_TARGET) $(NODE_RUNTIME) $(NODE_ABI_TARGET) $(NODE_DIST_URL) package $(NODE_BUILD_VERBOSE)
 
 npm-pack: clean package.json index.js
-	npm install node-pre-gyp@0.14.x
+	npm install --prefix=$(NPM_ROOT)/../ node-pre-gyp@0.14.x typescript@3.6.2 @types/argparse@1.0.38
+	tsc --version
+	tsc --outDir dist/ client.ts || true
```

Comment:
You can remove the flags here and instead update the `tsconfig.json` (see my comment below).

Reply:
As much as I could test, this was not working for me.

Reply:
@piraka9011 How are we suppsed to call `tsc` ? Using just `tsc` results in:
```
tsc || true
index.d.ts:32:14 - error TS2300: Duplicate identifier 'Stream'.

32 export class Stream {}
                ~~~~~~

  ../../../../../../node_modules/deepspeech/index.d.ts:32:14
    32 export class Stream {}
                    ~~~~~~
    'Stream' was also declared here.

index.d.ts:41:14 - error TS2300: Duplicate identifier 'Model'.

41 export class Model {
                ~~~~~

  ../../../../../../node_modules/deepspeech/index.d.ts:41:14
    41 export class Model {
                    ~~~~~
    'Model' was also declared here.

../../../../../../node_modules/deepspeech/index.d.ts:32:14 - error TS2300: Duplicate identifier 'Stream'.

32 export class Stream {}
                ~~~~~~

  index.d.ts:32:14
    32 export class Stream {}
                    ~~~~~~
    'Stream' was also declared here.

../../../../../../node_modules/deepspeech/index.d.ts:41:14 - error TS2300: Duplicate identifier 'Model'.

41 export class Model {
                ~~~~~

  index.d.ts:41:14
    41 export class Model {
                    ~~~~~
    'Model' was also declared here.


Found 4 errors.
```

Reply:
And running `tsc client.ts` overwrites existing `client.js` instead of producing `dist/client.js` (even though I have `outDir` in teh config file.

Reply:
What `tsc` version do you have? Can you try `npx tsc` instead?

Reply:
> What `tsc` version do you have? Can you try `npx tsc` instead?

3.6.2.

Reply:
> Can you try `npx tsc` instead?

Why ? What is it going to do differently ?

Reply:
Weird:
```
$ npx tsc --version
npx : 1 installé(s) en 0.776s
message TS6029: Version 1.5.3
$ PATH=$HOME/node_modules/.bin/:$PATH tsc --version
Version 3.6.2
```

Reply:
`npx tsc` uses the executables/modules installed in the local `node_modules` as opposed to the global namespace (so if you install a package with `npm install -g <package>` it is installed globally.

Reply:
`npx tsc` seems to install deprecated package: https://www.npmjs.com/package/tsc

Reply:
Once it loads the properly installed version from `typescript@3.6.2` package, same:
```
$ PATH=$HOME/node_modules/.bin/:$PATH npx tsc
index.d.ts:32:14 - error TS2300: Duplicate identifier 'Stream'.

32 export class Stream {}
                ~~~~~~

  ../../../../../../node_modules/deepspeech/index.d.ts:32:14
    32 export class Stream {}
                    ~~~~~~
    'Stream' was also declared here.

index.d.ts:41:14 - error TS2300: Duplicate identifier 'Model'.

41 export class Model {
                ~~~~~

  ../../../../../../node_modules/deepspeech/index.d.ts:41:14
    41 export class Model {
                    ~~~~~
    'Model' was also declared here.

../../../../../../node_modules/deepspeech/index.d.ts:32:14 - error TS2300: Duplicate identifier 'Stream'.

32 export class Stream {}
                ~~~~~~

  index.d.ts:32:14
    32 export class Stream {}
                    ~~~~~~
    'Stream' was also declared here.

../../../../../../node_modules/deepspeech/index.d.ts:41:14 - error TS2300: Duplicate identifier 'Model'.

41 export class Model {
                ~~~~~

  index.d.ts:41:14
    41 export class Model {
                    ~~~~~
    'Model' was also declared here.


Found 4 errors.

```

Reply:
Wait ... `../../../../../../node_modules/deepspeech/index.d.ts` That's my locally installed module ... Looks like it clashes definition ...

Reply:
I've removed the $HOME/node_modules/deepspeech/ and I'm back at square one. `tsc`, `tsc -p .` or `npx tsc` runs, output nothing on stderr/stdout and there is no `dist/` produced.

Reply:
`strace -ff tsc` shows it's correctly opening the `tsconfig.json` ... But nothing. No processing, no error. There's no verbose / debug mode either ?

Reply:
```
$ PATH=$HOME/node_modules/.bin/:$PATH tsc --build . --verbose
[18:17:30] Projects in this build: 
    * tsconfig.json

[18:17:30] Project 'tsconfig.json' is out of date because output file 'dist/client.js' does not exist

[18:17:30] Building project '/home/alex/codaz/Mozilla/DeepSpeech/deepspeech-kdavis/native_client/javascript/tsconfig.json'...

[18:17:33] Updating unchanged output timestamps of project '/home/alex/codaz/Mozilla/DeepSpeech/deepspeech-kdavis/native_client/javascript/tsconfig.json'...
```

Reply:
Per `strace` it also reads `index.d.ts` and `client.ts`, but no `dist/` is generated. And if I create the directory, it does not get populated.

Reply:
@piraka9011 As you can see, it reads everything but produces nothing ...

Reply:
Why do we have `"'noEmit": "true"` ? removing that it generates `dist/client.js` properly.